lib/activerecord-mysql-structure/active_record/structure_sql_sanitizer.rb

```diff
@@ -5,6 +5,9 @@ module ActiveRecordMySqlStructure
   class StructureSqlSanitizer
     COLUMN_REGEX = /\A\s*`\w+`/
 
+    # Naive index check. Covers the most common use-cases
```

Comment:
Hm should we mention what cases it doesn't cover? Just strikes me as one of those things where a developer wants to use something new and we end up with the same issue again but no one knows why. Just thinking adding some key terms that might show up in someone's search might be helpful to someone debugging an issue here.

Reply:
There's a few permutations. Per the `create_definition` clause [in the mysql documentation](https://dev.mysql.com/doc/refman/8.0/en/create-table.html) even the leading "KEY" is technically optional; if we wanted to just cover the _likely_ cases, I think we can get away with `/(\A\s*(?:(?:UNIQUE|FULLTEXT|SPATIAL) )?(KEY|INDEX)\s+.*?)(,?)(\s*\Z)/` if I understand the docs right, but that's not exhaustive. I did briefly humor the idea of finding a full blown mysql parser in ruby to cover our bases, but nothing especially practical turned up. 

Since this isn't searchable as part of our project anyway, I'm wondering if it might instead be helpful for the sanitizer to include a comment at the top of its regenerated `structure.sql` indicating that the gem rewrote the file?

Reply:
Yes, I do like that idea of the comment, that could definitely help someone zero in on the problem quickly.